libcaf_core/caf/detail/to_trait.hpp

```diff
@@ -0,0 +1,29 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#pragma once
+
+#include "caf/statically_typed.hpp"
+
+#include <type_traits>
+
+namespace caf::detail {
+
+template <class, bool>
+struct to_trait_oracle;
+
+template <class T>
+struct to_trait_oracle<T, true> {
+  using type = statically_typed<T>;
+};
+
+template <class T>
+struct to_trait_oracle<T, false> {
+  using type = T;
+};
+
+template <class T>
+using to_trait_t = typename to_trait_oracle<T, std::is_function_v<T>>::type;
```

Comment:
`trait` is way too generic here as a name. Especially without context in its own file. `to_signatures_wrapper_t` maybe? Plus, please leave a documentation here what's is used for.

Reply:
Will do. Not finding the right name here. Remember saying that `signatures_wrapper` isn't a good name, but it sounds better than `statically_typed_trait` though. Any opinions?